client/src/components/settings/Username.js

```diff
@@ -77,7 +78,11 @@ class UsernameSettings extends Component {
       /* eslint-disable-next-line react/no-did-update-set-state */
       return this.setState({
         isFormPristine: username === formValue,
-        submitClicked: false
+        submitClicked: false,
+        // eslint-disable-next-line max-len
+        isUserNew: /fcc([0-9|a-f]){8}-([0-9|a-f]){4}-([0-9|a-f]){4}-([0-9|a-f]){4}-([0-9|a-f]){12}$/g.test(
```

Comment:
```suggestion
        isUserNew: /^fcc[0-9|a-f]{8}-[0-9|a-f]{4}-[0-9|a-f]{4}-[0-9|a-f]{4}-[0-9|a-f]{12}$/.test(
```
The regex can be tightened up slightly.  Also, how about putting it in a const, since it would be appearing in two places?

Reply:
I'm wondering whether we even need this second test since we can just do it in the state declaration

Reply:
After a bit of poking around, I don't think we need the FIRST declaration because we don't have `prevUsername` over there. In addition, I think prevUsername is just what was in the component before it updated, not the username of the user I guess

Reply:
@ojeytonwilliams Are you sure the regex should not be the following?
```js
        isUserNew: /^fcc[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/.test(
```
and if so, I suggest creating the following regex above  the `UsernameSettings` class and `componentDidUpdate`:
```js
const hex = '[0-9a-f]';
const tempUserRegex = new RegExp(`^fcc${hex}{8}-\(${hex}{4}-\){3}${hex}{12}$`);
```
and replacing line 64 and 85 with:
```js
        isUserNew: tempUserRegex.test(prevUsername)
```

Reply:
That looks like a correct Regex too, I'll update it

Reply:
Look at my last edit, The `|` within the original regex would have not captured things correctly.